src/hidimstat/knockoffs.py

```diff
@@ -29,102 +84,76 @@ class ModelXKnockoff(BaseVariableImportance):
 
     Parameters
     ----------
-    X : 2D ndarray (n_samples, n_features)
-        The design matrix.
-
-    y : 1D ndarray (n_samples, )
-        The target vector.
-
-    estimator : sklearn estimator instance or a cross validation instance
-        The estimator used for fitting the data and computing the test statistics.
-        This can be any estimator with a `fit` method that accepts a 2D array and
-        a 1D array, and a `coef_` attribute that returns a 1D array of coefficients.
-        Examples include LassoCV, LogisticRegressionCV, and LinearRegression.
-
-        *Configuration example:*
-
-        | LassoCV(alphas=alphas, n_jobs=None, verbose=0, max_iter=1000,
-        | cv=KFold(n_splits=5, shuffle=True, random_state=0), tol=1e-8)
-
-        | LogisticRegressionCV(penalty="l1", max_iter=1000, solver="liblinear",
-        | cv=KFold(n_splits=5, shuffle=True, random_state=0), n_jobs=None,
-        | tol=1e-8)
-
-        | LogisticRegressionCV(penalty="l2", max_iter=1000, n_jobs=None,
-        | verbose=0, cv=KFold(n_splits=5, shuffle=True, random_state=0),
-        | tol=1e-8,)
-
-    preconfigure_estimator : callable, default=preconfigure_estimator_LassoCV
-        A function that configures the estimator for the Model-X knockoff procedure.
-        If provided, this function will be called with the estimator, X, X_tilde, and y
-        as arguments, and should modify the estimator in-place.
-
-    fdr : float, default=0.1
-        The desired controlled False Discovery Rate (FDR) level.
-
+    ko_generator : object
+        Knockoff generator implementing fit(X) and sample(n_repeats, random_state).
+    n_repeat : int, default=1
+        Number of knockoff draws to average over.
     centered : bool, default=True
-        Whether to standardize the data before performing the inference procedure.
-
-    cov_estimator : estimator object, default=LedoitWolf()
-        Estimator for empirical covariance matrix.
-
+        If True, standardize X before fitting the generator and computing statistics.
+    test_linear_model : estimator, default=LassoCV(...)
```

Comment:
```suggestion
    estimator: estimator, default=LassoCV(...)
```
I suggest renaming this `estimator` for consistency with all other methods from the API 2.

Reply:
Ues, it should be called `estimator` for the sake of homogeneity.